File: c471.c
#include <stdio.h>
#include <stdlib.h>

typedef long long ll;

int cmp(const void*, const void*);
ll printEnergy(int items[100000][2], int);

int main(void) {

    int N, i;
    int items[100000][2];
    
    while(scanf("%d", &N) != EOF) {
        for(i = 0;i < N;i++) scanf("%d", &items[i][0]);
        for(i = 0;i < N;i++) scanf("%d", &items[i][1]);
        qsort(items, N, sizeof(int)*2, cmp);
        printf("%lld\n", printEnergy(items, N));
    }

    return 0;

}

int cmp(const void* a, const void* b) {
    return ((int*)a)[0] * ((int*)b)[1] - ((int*)b)[0] * ((int*)a)[1];
}

ll printEnergy(int items[100000][2], int N) {

    ll energy = 0, weight = 0;

    for(int i = 1;i < N;i++) {

        weight += items[i - 1][0];
        energy += weight * items[i][1];

    }

    return energy;

}

File: c471/c471.py
def qsort(L:list, i: int, j: int):
    if j <= i: return
    start, end = i, j
    while i != j:
        while (L[start][0] * L[j][1] <= L[j][0] * L[start][1]) and (j > i): j -= 1
        while (L[start][0] * L[i][1] >= L[i][0] * L[start][1]) and (i < j): i += 1
        L[i], L[j] = L[j], L[i]

    L[start], L[i] = L[i], L[start]
    qsort(L, start, i - 1), qsort(L, i + 1, end)
    
def printEnergy(L: list):
    energy, weight = 0, 0
    for i in range(1, len(L)):
        weight += L[i - 1][0]
        energy += weight * L[i][1]
        
    print(energy)

def main():
    while True:
        try:
            N = int(input())
        except:
            break
        
        w = list(map(int, input().split()))
        f = list(map(int, input().split()))
        items = [[x, y]for x, y in zip(w, f)]
        qsort(items, 0, len(items) - 1)
        printEnergy(items)
    
    
main()
